### PR TITLE
[feat] NormalSSD의 read 기능 구현 및 테스트 추가

### PR DESCRIPTION
**Pull request by jihoonlee91**
## 📝 작업 요약

- NormalSSD의 read 테스트 추가
- NormalSSD의 read 기능 구현

---

### 🔍 코드 및 테스트 확인
- [x] 코드가 정상적으로 동작함
- [x] 필요한 테스트를 추가했고 기존 테스트를 통과함
- [x] 관련 문서를 업데이트함 (필요한 경우)
- [x] 리뷰어가 이해할 수 있도록 설명을 충분히 작성함

### 📦 PR 운영 규칙
- [x] 커밋 메시지와 브랜치 이름 규칙을 따름
- [ ] 2명 이상에게 Approve를 받음
- [ ] Merge는 PR 작성자가 직접 수행함
- [ ] Merge 후 PR을 Close하고 브랜치를 삭제함

---

## 💬 참고 사항

- 추후 write 구현이 완료되면 병합 및 리팩토링 예정

### CODE REVIEW

File: ssd/normal_ssd.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    EMPTY_VALUE: str = "0x00000000"
+    DEFAULT_LBA_RANGE: range = range(0, 100)
+
+    def __init__(
+        self,
+        nand_file: str = DEFAULT_NAND_FILE,
+        output_file: str = DEFAULT_OUTPUT_FILE,
+        lba_range: range = DEFAULT_LBA_RANGE,
+    ) -> None:
+        self.nand_file: str = nand_file
```
Comment (hkmilk0829) [inline ssd/normal_ssd.py:17]: 멤버 변수의 naming에  protected 변수로의 역할을 위해 언더바를 붙이는 것은 어떨까요?!

File: tests/test_ssd_read.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        if os.path.exists(f):
+            os.remove(f)
+
+
+def read_output() -> str:
+    with open(OUTPUT_FILE, "r") as f:
+        return f.read().strip()
+
+
+@pytest.mark.parametrize("initial_nand, read_lba, expected_output", [
```
Comment (hkmilk0829) [inline tests/test_ssd_read.py:26]: test_read()의 인자로 들어가는 변수명을 이해하기 쉽게 naming 해주셨고 테스트 케이스들을 다양하게 작성해주셨네요. 💯

**Review by hkmilk0829 — APPROVED** [review]

File: ssd/normal_ssd.py (hunk 1 above)
Comment (jihoonlee91) [reply ssd/normal_ssd.py:17]: 거기까지는 생각을 못했는데 중요한 부분 짚어주셔서 감사합니다. 바로 반영하겠습니다!

File: ssd/normal_ssd.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                if int(lba_str) == address:
+                    self._write_output(value)
+                    return
+
+        self._write_output(self.EMPTY_VALUE)
+
+    def write(self, address: int, value: int) -> None:
+        pass
+
+    def _write_output(self, value: str) -> None:
```
[the remark this answers is not in the document]
Comment (jihoonlee91) [reply ssd/normal_ssd.py:41]: 넵 write 메서드 구현 과정에서 형태를 바꿀 필요가 있으면 자유롭게 구현 부탁드립니다~😀

File: tests/test_ssd_read.py (hunk 2 above)
Comment (jihoonlee91) [reply tests/test_ssd_read.py:26]: 최대한 모든 edge case를 다 포함하려고 했는데, 보시다가 혹시 놓친 부분이 있다면 알려주시면 감사하겠습니다 ㅎㅎ

**Review by hkmilk0829 — APPROVED** [review]

File: tests/test_ssd_read.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import os
+import pytest
+from ssd.normal_ssd import NormalSSD
+
+NAND_FILE = "ssd_nand.txt"
+OUTPUT_FILE = "ssd_output.txt"
+
+
+@pytest.fixture(autouse=True)
```
Comment (jiyoung61) [inline tests/test_ssd_read.py:9]: `autouse=True` 기능은 어떤 기능인지 알 수 있을까요? 처음 보는 옵션이라 질문드립니다.

Comment (jihoonlee91) [reply tests/test_ssd_read.py:9]: autouse=True를 지정하면 해당 fixture가 테스트 함수에 명시적으로 넘겨지지 않아도 자동으로 실행됩니다.
즉, test_something(clean_files):처럼 인자로 넣지 않아도 자동으로 setup/teardown이 됩니다!

Comment (jiyoung61) [reply tests/test_ssd_read.py:9]: 감사합니다! 명시적 호출이 없는 리소스에 대해 setup/teardown 할 때 유용하게 사용할 수 있을 것 같습니다 👍

File: tests/test_ssd_read.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    ({"10": "0x11111111"}, 11, "0x00000000"),
+
+    # Empty NAND
+    ({}, 0, "0x00000000"),
+    ({}, 99, "0x00000000"),
+
+    # Invalid LBA range
+    ({"3": "0xAAAABBBB"}, -1, "ERROR"),
+    ({"3": "0xAAAABBBB"}, 100, "ERROR"),
+])
```
Comment (jiyoung61) [inline tests/test_ssd_read.py:49]: edeg case에 대한 테스트 케이스가 적절하게 추가 되었습니다 💯

**Review by jiyoung61 — APPROVED** [review]

File: tests/test_ssd_read.py (hunk 4 above)
Comment (KangSooHan) [reply tests/test_ssd_read.py:9]: 아니 이런 꿀기능이 있다니..! 꿀팁 감사합니다 👍
